File: problem/Simple_Bank_System/Bank.py
class Bank:
    def __init__(self, balance):
        self.bal = balance
        self.n = len(balance)

    def valid(self, acc):
        return 1 <= acc <= self.n

    def transfer(self, account1, account2, money):
        if not self.valid(account1) or not self.valid(account2) or self.bal[account1 - 1] < money:
            return False
        self.bal[account1 - 1] -= money
        self.bal[account2 - 1] += money
        return True

    def deposit(self, account, money):
        if not self.valid(account):
            return False
        self.bal[account - 1] += money
        return True

    def withdraw(self, account, money):
        if not self.valid(account) or self.bal[account - 1] < money:
            return False
        self.bal[account - 1] -= money
        return True

File: problem/Simple_Bank_System/Bank.js
var Bank = function(balance) {
    this.bal = balance;
    this.n = balance.length;
};

Bank.prototype.valid = function(acc) {
    return acc > 0 && acc <= this.n;
};

Bank.prototype.transfer = function(account1, account2, money) {
    if (!this.valid(account1) || !this.valid(account2) || this.bal[account1 - 1] < money)
        return false;
    this.bal[account1 - 1] -= money;
    this.bal[account2 - 1] += money;
    return true;
};

Bank.prototype.deposit = function(account, money) {
    if (!this.valid(account))
        return false;
    this.bal[account - 1] += money;
    return true;
};

Bank.prototype.withdraw = function(account, money) {
    if (!this.valid(account) || this.bal[account - 1] < money)
        return false;
    this.bal[account - 1] -= money;
    return true;
};